lib/reversetunnel/agentpool.go

```diff
@@ -531,6 +531,14 @@ func (p *AgentPool) transport(ctx context.Context, channel ssh.Channel, requests
 		proxySigner:          p.PROXYSigner,
 		forwardClientAddress: true,
 	}
+
+	if p.IsRemoteCluster {
+		if tracker, ok := p.ReverseTunnelServer.(interface{ TrackUserConnection() func() }); ok {
+			t.trackUserConnection = tracker.TrackUserConnection
+		}
```

Comment:
I find this difficult to parse at a glance... anonymous interface declaring a function that returns a function inside an assignment inside an if statement is just one too many things I think.  I'd suggest declaring this interface separately.

Additionally, I think `TrackUserConnection` is the kind of function that really benefits from named return types:

```golang
type userConnectionTracker interface {
    TrackUserConnection() (cancel func())
}
```

intended usage pattern is much more obvious IMO.